node/impl/full/eth.go

```diff
@@ -672,6 +674,42 @@ func (a *EthModule) EthChainId(ctx context.Context) (ethtypes.EthUint64, error)
 	return ethtypes.EthUint64(build.Eip155ChainId), nil
 }
 
+func (a *EthModule) EthSyncing(ctx context.Context) (ethtypes.EthSyncingResult, error) {
+	state, err := a.SyncAPI.SyncState(ctx)
+	if err != nil {
+		return ethtypes.EthSyncingResult{}, fmt.Errorf("failed calling SyncState: %w", err)
+	}
+
+	if len(state.ActiveSyncs) == 0 {
+		return ethtypes.EthSyncingResult{}, errors.New("no active syncs, try again")
+	}
+
+	working := -1
+	for i, ss := range state.ActiveSyncs {
+		if ss.Stage == api.StageIdle {
+			continue
+		}
+		working = i
```

Comment:
Can you explain this? Why are looking for the index of the last non-idle one from ActiveSyncs and eg not the 1st one?

Reply:
I wanted to have the same logic as when calling `lotus sync wait` which is done like this (see https://github.com/filecoin-project/lotus/blob/master/cli/sync.go#L286)

There is also [this commit](https://github.com/filecoin-project/lotus/pull/4599/commits/d6e2c80608179fa7abc488889fee1478fc0d1493) which picks the last worker in case all are idle, but it does not go in details as to why